### PR TITLE
chore: add e2e test for reconnect

### LINKED ISSUE #592: [flagd] implement reconnect gherkin test
Implement [this](https://github.com/open-feature/flagd-testbed/blob/main/gherkin/flagd-reconnect.feature).

Self-assigning.

### PR DESCRIPTION
**Pull request by toddbaert**
* adds reconnect e2e test for flagd RPC and in-process providers
* same as what's implemented in JS contribs
* cleanup in unrelated tests

This is stretching the ability of gherkin to describe test scenarios, but it's still a nice way of making sure this is implemented everywhere.

Fixes: https://github.com/open-feature/java-sdk-contrib/issues/592

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,16 +12,26 @@ jobs:
...
+          - 8014:8013
+      # sync-testbed for flagd in-process provider e2e tests
       sync:
         image: ghcr.io/open-feature/sync-testbed:v0.4.9
         ports:
           - 9090:9090
+      # sync-testbed for flagd in-process provider reconnect e2e tests
+      sync-unstable:
+        image: ghcr.io/open-feature/sync-testbed-unstable:v0.4.9
+        ports:
```
Comment (toddbaert) [inline .github/workflows/ci.yml]: The flagd in these containers restart every 5s, Giving use a nice way to test reconnection.

File: providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/reconnect/process/FlagdInProcessSetup.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private static FeatureProvider provider;
+    
+    @BeforeAll()
+    public static void setup() throws InterruptedException {
+        FlagdInProcessSetup.provider = new FlagdProvider(FlagdOptions.builder()
+        .resolverType(Config.Evaluator.IN_PROCESS)
+        .deadline(3000)
+        .host("localhost")
+        .port(9091)
+        .build());
```
Comment (toddbaert) [inline providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/reconnect/process/FlagdInProcessSetup.java:26]: connect to the "unstable" container.

File: providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/reconnect/steps/StepDefinitions.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        Awaitility.await().atMost(Duration.ofSeconds(10))
+        .until(() -> {
+            return this.errorHandlerRunCount > 0;
+        });
+    }
+
+    @Then("when the connection is reestablished the PROVIDER_READY handler must run again")
+    public void when_the_connection_is_reestablished_the_provider_ready_handler_must_run_again() {
+        // wait up to 10 seconds for a reconnect (PROVIDER_READY event)
+        Awaitility.await().atMost(Duration.ofSeconds(10))
```
Comment (toddbaert) [inline providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/reconnect/steps/StepDefinitions.java:98]: Awaitility is great for this.

File: providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/reconnect/steps/StepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        Awaitility.await().atMost(Duration.ofSeconds(15))
+        .until(() -> {
+            return this.errorHandlerRunCount > 0;
+        });
+    }
+
+    @Then("when the connection is reestablished the PROVIDER_READY handler must run again")
+    public void when_the_connection_is_reestablished_the_provider_ready_handler_must_run_again() {
+        // wait up to 15 seconds for a reconnect (PROVIDER_READY event)
+        Awaitility.await().atMost(Duration.ofSeconds(15))
```
Comment (toddbaert) [inline providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/reconnect/steps/StepDefinitions.java:101]: Awaitility is great for this. We wait up to 15s, but it's usually much quicker than that (more like 2.5s).

File: providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/FlagdProviderTest.java [hunk 5]
```diff
@@ -26,6 +25,7 @@
 import dev.openfeature.sdk.Reason;
 import dev.openfeature.sdk.Structure;
 import dev.openfeature.sdk.Value;
+import io.cucumber.java.AfterAll;
```
Comment (toddbaert) [inline providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/FlagdProviderTest.java:28]: Changes in this file are not related, just doing some clenaup of unused vars and reducing log noise.

**Review by beeme1mr — APPROVED** [review]
